### PR TITLE
WebSocketUpgradeHandler should not require ContextHandler

### PR DESCRIPTION
**Pull request by lachlan-roberts**
`WebSocketUpgradeHandler` should not be required to be wrapped inside a `ContextHandler`

### CODE REVIEW

**Comment by sbordet** [comment]
@lachlan-roberts can you explain why you need this? I'm not against it, just curious about why.
Please add the project this PR belongs to (and an issue if any).

**Comment by lachlan-roberts** [comment]
I don't specifically need this and I did not create an issue for it. I just saw these tests and didn't think the `WebSocketUpgradeHandler` should be forced to be under a `ContextHandler`. 

We don't require a `ContextHandler` for the websocket-core upgrade handler.

File: jetty-core/jetty-websocket/jetty-websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/WebSocketServerComponents.java [hunk 1]
```diff
@@ -62,28 +64,58 @@ public class WebSocketServerComponents extends WebSocketComponents
...
+     * Server attributes can be set with {@link #WEBSOCKET_BUFFER_POOL_ATTRIBUTE}, {@link #WEBSOCKET_INFLATER_POOL_ATTRIBUTE}
+     * and {@link #WEBSOCKET_DEFLATER_POOL_ATTRIBUTE} to override the {@link ByteBufferPool}, {@link DeflaterPool} or
+     * {@link InflaterPool} used by the components, otherwise this will try to use the pools shared on the {@link Server}.
+     * </p>
+     * @param server the server.
+     * @return the WebSocketComponents that was created or found.
+     */
+    public static WebSocketComponents ensureWebSocketComponents(Server server)
+    {
+        return ensureWebSocketComponents(server, server, server);
```
Comment (sbordet) [inline jetty-core/jetty-websocket/jetty-websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/WebSocketServerComponents.java]: For symmetry, 2nd parameter shouldn't be `server.getContext()`?

File: jetty-core/jetty-websocket/jetty-websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/WebSocketServerComponents.java [hunk 2]
```diff
@@ -126,7 +158,16 @@ public String toString()
...
+        WebSocketComponents components = (WebSocketComponents)contextHandler.getContext().getAttribute(WEBSOCKET_COMPONENTS_ATTRIBUTE);
+        if (components == null)
+            throw new IllegalStateException("WebSocketComponents has not been created");
+
+        return components;
+    }
+
+    public static WebSocketComponents getWebSocketComponents(Server server)
+    {
+        WebSocketComponents components = (WebSocketComponents)server.getAttribute(WEBSOCKET_COMPONENTS_ATTRIBUTE);
```
Comment (sbordet) [inline jetty-core/jetty-websocket/jetty-websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/WebSocketServerComponents.java]: Refactor both methods to call a private method that takes `Attributes` as parameter?

File: jetty-core/jetty-websocket/jetty-websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/ServerWebSocketContainer.java [hunk 3]
```diff
@@ -106,6 +106,11 @@ public static ServerWebSocketContainer get(Context context)
     private final FrameHandlerFactory factory;
     private InvocationType invocationType = InvocationType.BLOCKING;
 
+    public ServerWebSocketContainer()
+    {
+        this(new WebSocketMappings());
+    }
+
```
Comment (sbordet) [inline jetty-core/jetty-websocket/jetty-websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/ServerWebSocketContainer.java]: I'd remove this, I prefer to keep instantiation through static methods.

File: jetty-core/jetty-websocket/jetty-websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/WebSocketUpgradeHandler.java [hunk 4]
```diff
@@ -93,8 +93,29 @@ public static WebSocketUpgradeHandler from(Server server, ContextHandler context
...
+        return new WebSocketUpgradeHandler(container);
+    }
+
     private final ServerWebSocketContainer _container;
 
+    public WebSocketUpgradeHandler()
+    {
+        this(new ServerWebSocketContainer());
+    }
+
```
Comment (sbordet) [inline jetty-core/jetty-websocket/jetty-websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/WebSocketUpgradeHandler.java]: I'd remove this too, prefer the static methods.

**Review by sbordet — CHANGES_REQUESTED** [review]

**Review by sbordet — DISMISSED** [review]
@lachlan-roberts I have updated WebSocketUpgradeHandler to expose the ServerWebSocketContainer as attribute also when created with only the Server.
LGTM now.

**Review by sbordet — APPROVED** [review]

**Pull request merged**